crates/store/src/state.rs

```diff
@@ -922,14 +1018,14 @@ impl State {
     ) -> Result<AccountProofResponse, DatabaseError> {
         let AccountProofRequest { block_num, account_id, details } = account_request;
 
-        if details.is_some() && !account_id.has_public_state() {
+        if details.is_some() && !account_id.is_public() {
             return Err(DatabaseError::AccountNotPublic(account_id));
         }
```

Comment:
Why did we change `is_public()` to `has_public_state()` here? `is_public()` returns false for network accounts - so, `has_public_state()` is correct here.